### PR TITLE
Nuget update

### PR DESCRIPTION
**Pull request by ErikApption**
- new MS Graph client which has a totally different API - so had to rewrite all the calls. 
- Other updates were more minor.
- cleanup transitive dependencies using NugetNinja

### CODE REVIEW

File: Portal/src/Datahub.Core/Datahub.Core.csproj [hunk 1]
```diff
@@ -23,62 +23,63 @@
 		<PackageReference Include="Ae.MudBlazor.Forms" Version="0.5.22329.4-beta" />
 		<PackageReference Include="AElemental.Blazor" Version="1.6.22173.2" />
 		<PackageReference Include="AElemental.Blazor.Theme" Version="1.6.22173.2" />
-		<PackageReference Include="AngleSharp" Version="1.0.0-alpha-844" />
+		<PackageReference Include="AngleSharp" Version="1.0.4" />
```
Comment (yjmrobert) [inline Portal/src/Datahub.Core/Datahub.Core.csproj:26]: Curious where this is used in Core

File: Portal/src/Datahub.Core/Datahub.Core.csproj [hunk 2]
```diff
@@ -23,62 +23,63 @@
...
 		<PackageReference Include="BlazorInputFile" Version="0.2.0" />
-		<PackageReference Include="BlazorTemplater" Version="1.5.0" />
+		<PackageReference Include="BlazorTemplater" Version="1.5.1" />
 		<PackageReference Include="ByteSize" Version="2.1.1" />
-		<PackageReference Include="DeepL" Version="0.4.2" />
-		<PackageReference Include="Foundatio" Version="10.5.0" />
-		<PackageReference Include="Foundatio.AzureStorage" Version="10.5.0" />
+		<PackageReference Include="DeepL.net" Version="1.7.1" />
+		
+		<PackageReference Include="Foundatio.AzureStorage" Version="10.6.1" />
```
Comment (yjmrobert) [inline Portal/src/Datahub.Core/Datahub.Core.csproj:39]: Do we still use foundatio?

File: Portal/src/Datahub.Core/Datahub.Core.csproj [hunk 3]
```diff
@@ -89,55 +90,55 @@
...
-		<GitHookFiles Include="../.githooks/*.*" />
+		<GitHookFiles Include="../.githooks/*.*"></GitHookFiles>
 	</ItemGroup>
 	<ItemGroup>
-	  <Compile Remove="Services\AzureCosting\**" />
-	  <Content Remove="Services\AzureCosting\**" />
-	  <EmbeddedResource Remove="Services\AzureCosting\**" />
-	  <GitHookFiles Remove="Services\AzureCosting\**" />
-	  <None Remove="Services\AzureCosting\**" />
+		<Compile Remove="Services\AzureCosting\**" />
```
Comment (yjmrobert) [inline Portal/src/Datahub.Core/Datahub.Core.csproj]: Why is this folder removed from all the steps?

File: Portal/src/Datahub.Core/Datahub.Core.csproj [hunk 4]
```diff
@@ -89,55 +90,55 @@
...
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-regular-400.ttf" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-regular-400.woff2" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-sharp-solid-900.ttf" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-sharp-solid-900.woff2" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-solid-900.ttf" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-solid-900.woff2" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-thin-100.ttf" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-thin-100.woff2" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-v4compatibility.ttf" />
+		<_ContentIncludedByDefault Remove="wwwroot\css\webfonts\fa-v4compatibility.woff2" />
```
Comment (yjmrobert) [inline Portal/src/Datahub.Core/Datahub.Core.csproj]: I think we can remove all of these and put them in the portal project

File: Portal/src/Datahub.Core/Services/TranslationService.cs [hunk 5]
```diff
@@ -3,32 +3,33 @@
...
 {
-    private readonly DeepLClient _client;
+    private readonly Translator _client;
 
     public TranslationService(IConfiguration configuration)
     {
         var useFreeApi = configuration.GetSection("DeepL").GetValue<bool>("UseFreeApi");
         var authKey = configuration.GetSection("DeepL").GetValue<string>("AuthKey");
-        _client = new DeepLClient(authKey, useFreeApi);
+        _client = new Translator(authKey, new TranslatorOptions() { ServerUrl = useFreeApi ? "https://api-free.deepl.com" : null });
```
Comment (yjmrobert) [inline Portal/src/Datahub.Core/Services/TranslationService.cs:17]: Can we slip this into the config object as a default value

File: Portal/utils/SyncDocs/Program.cs [hunk 6]
```diff
@@ -44,10 +45,9 @@
 
         // delete unnecessary files
         fileMappingService.CleanUpMappings();
-
+        if (string.IsNullOrWhiteSpace(deeplKey)) throw new InvalidOperationException($"Deepl Key is missing");
```
Comment (yjmrobert) [inline Portal/utils/SyncDocs/Program.cs:48]: 💗

**Review by yjmrobert — APPROVED** [review]

File: Portal/src/Datahub.Core/Datahub.Core.csproj (hunk 2 above)
Comment (ErikApption) [reply Portal/src/Datahub.Core/Datahub.Core.csproj:39]: nope - its upload features are too limited

File: Portal/src/Datahub.Core/Datahub.Core.csproj (hunk 1 above)
Comment (ErikApption) [reply Portal/src/Datahub.Core/Datahub.Core.csproj:26]: Good question

File: Portal/src/Datahub.Core/Datahub.Core.csproj (hunk 3 above)
Comment (ErikApption) [reply Portal/src/Datahub.Core/Datahub.Core.csproj]: there was a bunch of weird include/exclude in this project. Good catch, I did clean them up.

File: Portal/src/Datahub.Core/Services/TranslationService.cs (hunk 5 above)
Comment (ErikApption) [reply Portal/src/Datahub.Core/Services/TranslationService.cs:17]: We should resolve the deepl subscription first

File: Portal/utils/SyncDocs/Program.cs (hunk 6 above)
Comment (ErikApption) [reply Portal/utils/SyncDocs/Program.cs:48]: lol - this prevents to delete all the docs when there is no valid key

File: Portal/src/Datahub.Core/Datahub.Core.csproj (hunk 1 above)
Comment (ErikApption) [reply Portal/src/Datahub.Core/Datahub.Core.csproj:26]: Used in ExcelHtmlContentParser

**Pull request merged**